base/util/src/main/java/com/netscape/cmsutil/crypto/CryptoUtil.java

```diff
@@ -2576,30 +2656,53 @@ public static void importSharedSecret(byte[] wrappedSessionKey,byte[] wrappedSha
         CryptoManager cm = CryptoManager.getInstance();
         CryptoToken token = cm.getInternalKeyStorageToken();
 
-        KeyManager km = new KeyManager(token);
+        String method = "CryptoUtil.importSharedSecret ";
+        logger.debug(method + " nickname: " + sharedSecretNickname);
 
+        KeyManager km = new KeyManager(token);
         if (km.uniqueNamedKeyExists(sharedSecretNickname)) {
             throw new IOException("Shared secret " + sharedSecretNickname + " already exists");
         }
 
         //Unwrap session key
 
         KeyWrapper keyWrap = token.getKeyWrapper(KeyWrapAlgorithm.RSA);
+        KeyWrapper keyWrapOAEP = token.getKeyWrapper(KeyWrapAlgorithm.RSA_OAEP);
+        logger.debug(method + " subsytemCertNickname: " + subsystemCertNickname);
+        System.out.println(method + " subsytemCertNickname: " + subsystemCertNickname);
+
         X509Certificate cert = cm.findCertByNickname(subsystemCertNickname);
+        logger.debug(method + " subsystemCert: " + cert);
         PrivateKey subsystemPrivateKey = cm.findPrivKeyByCert(cert);
         keyWrap.initUnwrap(subsystemPrivateKey,null);
 
-        SymmetricKey unwrappedSessionKey = keyWrap.unwrapSymmetric(wrappedSessionKey, SymmetricKey.DES3,
+        SymmetricKey unwrappedSessionKey = null;
+        //Since we don't know if aep was used to wrap the key, try with and without.
+        try {
+            unwrappedSessionKey =  keyWrap.unwrapSymmetric(wrappedSessionKey, SymmetricKey.AES,
                 0);
-
-        SymmetricKey unwrappedSharedSecret = null;
+        } catch(Exception e) {
+            logger.debug(method + " exception found, trying RSA-OAEP: " + e);
```

Comment:
btw, these logger debug calls (here and at other places in CryptoUtil.java) probably aren't useful when being called by cli's such as when pkispawn is at work.  You might wat to consider using System.out.println().

Reply:
Done next PR.